src/Grassmann.jl

```diff
@@ -0,0 +1,275 @@
+"""
+struct Grassmann{N,K,T} <: Manifold end
+Grassmann(n::Int, k::Int, T::Type = Real) = Grassmann{n, k, T}()
+
+function check_manifold_point(G::Grassmann{N,K,T},x; kwargs...) where {N,K,T}
+    if (T <: Real) && !(eltype(x) <: Real)
+        return DomainError(eltype(x),
+            "The matrix $(x) is not a real-valued matrix, so it does noe lie on the Grassmann manifold of dimension ($(N),$(K)).")
+    end
+    if (T <: Complex) && !(eltype(x) <: Real) && !(eltype(x) <: Complex)
+        return DomainError(eltype(x),
+            "The matrix $(x) is neiter real- nor complex-valued matrix, so it does noe lie on the complex Grassmann manifold of dimension ($(N),$(K)).")
+    end
+    if any( size(x) != representation_size(G) )
+        return DomainError(size(x),
+            "The matrix $(x) is does not lie on the Grassmann manifold of dimension ($(N),$(K)), since its dimensions are wrong.")
+    end
+    c = x'*x
+    if !isapprox(c, one(c); kwargs...)
+        return DomainError(norm(c-one(c)),
+            "The point $(x) does not lie on the Grassmann manifold of dimension ($(N),$(K)), because x'x is not the unit matrix.")
+    end
+end
+function check_tangent_vector(G::Grassmann{N,K,T},x,v; kwargs...) where {N,K,T}
+    t = check_manifold_point(G,x)
+    if (t != nothing)
+        return t
+    end
+    if (T <: Real) && !(eltype(v) <: Real)
+        return DomainError(eltype(v),
+            "The matrix $(v) is not a real-valued matrix, so it can not be a tangent vector to the Grassmann manifold of dimension ($(N),$(K)).")
+    end
+    if (T <: Complex) && !(eltype(v) <: Real) && !(eltype(v) <: Complex)
+        return DomainError(eltype(v),
+            "The matrix $(v) is neiter real- nor complex-valued matrix, so it can not bea tangent vector to the complex Grassmann manifold of dimension ($(N),$(K)).")
+    end
+    if any( size(v) != representation_size(G) )
+        return DomainError(size(v),
+            "The matrix $(v) is does not lie in the tangent space of $(x) on the Grassmann manifold of dimension ($(N),$(K)), since its dimensions are wrong.")
+    end
+    if !isapprox(x'*v + v'*x, zeros(K,K); kwargs...)
+        return DomainError(norm(x'*v + v'*x),
+            "The matrix $(v) is does not lie in the tangent space of $(x) on the Grassmann manifold of dimension ($(N),$(K)), since x'v + v'x is not the zero matrix.")
+    end
+end
+
+@doc doc"""
+    distance(M,x,y)
+
+computes the Riemannian distance on [`Grassmann`](@ref) manifold `M`$= \mathrm{Gr}(n,k)$.
+
+Let $USV = {\bar x}^\mathrm{T}y$ denote the SVD decomposition of
+${\bar x}^\mathrm{T}y$, where ${\bar\cdot}^{\mathrm{T}}$ denotes the complex
+conjugate transposed. Then the distance is given by
+````math
+d_{\mathrm{GR}(n,k)}(x,y) = \operatorname{norm}(\operatorname{Re}(b)).
+````
+where
+
+$b_{i}=\begin{cases} 0 & \text{if} \; S_i≧1 \\ \operatorname{acos}(S_i) & \, \text{if} \; S_i<1 \end{cases}.$
+"""
+function distance(M::Grassmann{N,K,T}, x, y) where {N,K,T}
```

Comment:
There is something wrong with indentation in this function (there are some tabs instead of spaces).